applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -21,91 +21,101 @@
 
-namespace Kratos
-{
+namespace Kratos {
 
 template<class TSparseSpace, class TDenseSpace, class TLinearSolver>
-class SolvingStrategyFactory
-{
+class SolvingStrategyFactory {
 public:
-    [[nodiscard]] static std::unique_ptr<SolvingStrategy<TSparseSpace, TDenseSpace>> Create(const Parameters& rSolverSettings, ModelPart& rModelPart)
-    {
+    [[nodiscard]] static std::unique_ptr<SolvingStrategy<TSparseSpace, TDenseSpace>>
+    Create(const Parameters &rSolverSettings, ModelPart &rModelPart) {
         const std::string strategy_type = "strategy_type";
 
         KRATOS_ERROR_IF_NOT(rSolverSettings.Has(strategy_type))
-        << "The parameter strategy_type is undefined, aborting.";
+                    << "The parameter strategy_type is undefined, aborting.";
 
         const auto echo_level = rSolverSettings["echo_level"].GetInt();
 
-        auto solver = LinearSolverFactory<TSparseSpace, TDenseSpace>().Create(rSolverSettings["linear_solver_settings"]);
+        auto solver = LinearSolverFactory<TSparseSpace, TDenseSpace>().Create(
+                rSolverSettings["linear_solver_settings"]);
         KRATOS_ERROR_IF_NOT(solver) << "Failed to create a linear solver" << std::endl;
 
         auto scheme = SchemeFactory<TSparseSpace, TDenseSpace>::Create(rSolverSettings);
         KRATOS_ERROR_IF_NOT(scheme) << "Failed to create a scheme" << std::endl;
 
-        auto builder_and_solver = BuilderAndSolverFactory<TSparseSpace, TDenseSpace, TLinearSolver>::Create(rSolverSettings, solver);
+        auto builder_and_solver = BuilderAndSolverFactory<TSparseSpace, TDenseSpace, TLinearSolver>::Create(
+                rSolverSettings, solver);
         KRATOS_ERROR_IF_NOT(builder_and_solver) << "Failed to create a builder-and-solver" << std::endl;
         builder_and_solver->SetEchoLevel(echo_level);
 
         auto criteria = ConvergenceCriteriaFactory<TSparseSpace, TDenseSpace>::Create(rSolverSettings);
         KRATOS_ERROR_IF_NOT(criteria) << "Failed to create convergence criteria" << std::endl;
         criteria->SetEchoLevel(echo_level);
 
-        if (rSolverSettings[strategy_type].GetString() == "newton_raphson")
-        {
+        if (rSolverSettings[strategy_type].GetString() == "newton_raphson") {
             const auto max_iterations = rSolverSettings["max_iterations"].GetInt();
             const auto calculate_reactions = rSolverSettings["calculate_reactions"].GetBool();
             const auto reform_dof_set_at_each_step = rSolverSettings["reform_dofs_at_each_step"].GetBool();
             const auto move_mesh_flag = rSolverSettings["move_mesh_flag"].GetBool();
 
-            auto strategy_parameters = ExtractStrategyParameters(rSolverSettings);
+            const std::vector<std::string> strategy_entries = {"loads_sub_model_part_list",
+                                                               "loads_variable_list"};
+            auto strategy_parameters = ExtractStrategyParameters(rSolverSettings, strategy_entries);
```

Comment:
I expect this can be const 
```suggestion
            const auto strategy_parameters = ExtractStrategyParameters(rSolverSettings, strategy_entries);
```

Reply:
done

Reply:
In the end we found it couldn't be due to the strategy constructor